### PR TITLE
Fix #6202: Avoid breaking occupied benches

### LINKED ISSUE #6202: Guests are not ejected from Broken benches
**OS:** Any
**Version:** Any
**Commit/Build:** Any

If a guest is sitting on a bench when it is broken, the guest is not removed from the bench. Instead, they continue to sit there, until they decide to get up.

- [x] Reproducible in RCT2 (vanilla)?
- [x] Multiplayer?

**Steps to reproduce:**
1. Have guest on Bench
2. Another guest breaks the bench
3. The guests continues sitting on the broken bench.

### PR DESCRIPTION
**Pull request by oli414**
As per @ZehMatt's suggestion on https://github.com/OpenRCT2/OpenRCT2/issues/6202. Peeps will not break benches if the bench is occupied.

Other potential solution would be to eject the sitting peeps when the bench is broken. I personally think however that this fix is more logical.

### CODE REVIEW

File: src/openrct2/peep/peep.c [hunk 1]
```diff
@@ -5565,6 +5565,31 @@ static void peep_update_walking_break_scenery(rct_peep* peep){
     sint32 edges = map_element->properties.path.edges & 0xF;
     if (edges == 0xF) return;
 
+    uint16 sprite_id = sprite_get_first_in_quadrant(peep->x, peep->y);
+
+    // Check if a peep is already sitting on the bench.
+    for (rct_sprite* sprite; sprite_id != SPRITE_INDEX_NULL; sprite_id = sprite->unknown.next_in_quadrant)
```
Comment (Gymnasiast) [inline src/openrct2/peep/peep.c]: `rct_sprite * sprite`

File: src/openrct2/peep/peep.c [hunk 2]
```diff
@@ -5565,6 +5565,31 @@ static void peep_update_walking_break_scenery(rct_peep* peep){
     sint32 edges = map_element->properties.path.edges & 0xF;
     if (edges == 0xF) return;
 
+    uint16 sprite_id = sprite_get_first_in_quadrant(peep->x, peep->y);
+
+    // Check if a peep is already sitting on the bench.
```
Comment (Gymnasiast) [inline src/openrct2/peep/peep.c]: `// Check if a peep is already sitting on the bench. If so, do not vandalise it.`

File: src/openrct2/peep/peep.c [hunk 3]
```diff
@@ -5565,6 +5565,31 @@ static void peep_update_walking_break_scenery(rct_peep* peep){
...
+
+        if (sprite->peep.state != PEEP_STATE_SITTING)
+        {
+            continue;
+        }
+
+        if (peep->z != sprite->peep.z)
+        {
+            continue;
+        }
```
Comment (Gymnasiast) [inline src/openrct2/peep/peep.c]:
```
if ((sprite->unknown.linked_list_type_offset != SPRITE_LIST_PEEP * 2) ||
    (sprite->peep.state != PEEP_STATE_SITTING) ||
    (peep->z != sprite->peep.z))
{
    continue;
}
```

**Comment by oli414** [comment]
@janisozaur I'm not quite sure what needs to be done on the network side for this. Could you elaborate?

**Comment by Gymnasiast** [comment]
Looking good! Could you raise the network version and update the changelog?

**Comment by oli414** [comment]
Is this the right way to do it?

**Comment by janisozaur** [comment]
yes

**Comment by duncanspumpkin** [comment]
Wait why was this done. It should just evict the peeps if they vandalise the bench.

**Comment by duncanspumpkin** [comment]
With this change busy parks will never have vandalism of benches making the game easier.

**Comment by Gymnasiast** [comment]
> It should just evict the peeps if they vandalise the bench.

That makes little sense to me. Think about it: are vandals going to vandalise a bench with people sitting on it in real life? Only the worst of the vandals, perhaps. It just borders on assault.

**Comment by oli414** [comment]
Exactly. There were two options to solving the issue. The original behaviour is for sure unintended behaviour. How would Chris Sawyer have solved it? I do think from a realise standpoint this solution would make the most sense and would probably be what Chris would have done as well.

It is open for discussion though as I can totally see that there're two sides to this discussion.

**Comment by duncanspumpkin** [comment]
It makes sense that a vandal disrupts a park. This makes it easier the busier the park which is the opposite of what you would expect from a game.

**Pull request merged**